### DISCUSSION TITLE
Widgets for different sections of the Wave App

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#2310] [Q&A]
Hi!I have a wave dashboard style web app successfully deployed to production. Is it possible to have widgets for different sections of the Wave App? Like how we have widgets for Android or IOS etc. For example, I have two sections-Home Page and Insights which are as two separate tabs using `ui.tabs()` component. Is it possible for me to have a Home Page widget and an insights widget to directly view the different sections?

**Comment by mturoci** [comment]
Can you maybe add some screenshots to better show what you want to achieve? I am not sure after reading the ticket.

**Reply by n-srinidhi** [reply]
Hi, So my current app has following sections:
![WhatsApp Image 2024-04-10 at 10 50 37 AM](https://github.com/h2oai/wave/assets/117831190/361643ad-261d-4ef4-8aff-1290abd03b0b)
Where each tab displays graphs, tables and other wave components. When I say widgets, I'd like to make somethign like this:
![widget](https://github.com/h2oai/wave/assets/117831190/7faf2781-3dfd-4e0b-b0e7-c7feb86ad994)
So instead of accessing whole app, I can directly use widgets to view different sections of the app. So I'll have one widget for dashboard, that shows only say one graph or table from the dashboard, etc.

**Reply by mturoci** [reply]
Aren't your widgets just Wave cards?

<img width="913" alt="image" src="https://github.com/h2oai/wave/assets/64769322/4d0074c0-492e-4ef3-9421-9bacfd2a3099">

**Reply by n-srinidhi** [reply]
that would essentially be as a Wave app right?
I am looking for creating Widgets for sections of my app, Similar to how we have Widgets on Android or IOS on our smart phones.

**Reply by mturoci** [reply]
Where do you want the widgets to be? I thought we are talking about having a page in a Wave app with widgets section that will take the user to the corresponding part of the app. Or did I get it wrong?

**Reply by n-srinidhi** [reply]
So the setup is such that we have a wave app, hosted live mapped to a domain. This wave app is being absorbed as an `IFrame` into another existing dashboard. So, they would like widgets within their dashboard to the specific sections within the wave app i have built, as shared in the screenshot images above.

**Reply by mturoci** [reply]
It's clear now. In that case, the wrapping dashboard needs to build those widgets and simply route to the sections of your wave app on click via [hash routing](https://wave.h2o.ai/docs/routing/#hash-routing).

**Reply by n-srinidhi** [reply]
Will adding hash routing impact the existing to and fro navigation happening via `ui.tabs()` in my dashboard?

**Reply by mturoci** [reply]
You need to make your tabs work with hash routing in order to be linkable - navigating to a link opens the corresponding tab.
